polaris-react/.storybook/preview.js

```diff
@@ -6,6 +6,17 @@ import {GridOverlay} from './GridOverlay';
 import {RenderPerformanceProfiler} from './RenderPerformanceProfiler';
 import {gridOptions, featureFlagOptions} from './manager';
 import {breakpoints} from '@shopify/polaris-tokens';
+import isChromatic from 'chromatic/isChromatic';
+
+// Use the document.fonts API to ensure fonts have fully loaded before rendering a story. Important for consistency when doing visual diffing in Chromatic.
+// See: https://developer.mozilla.org/en-US/docs/Web/API/Document/fonts
+const fontLoader = async () => ({
+  fonts: await document.fonts.ready,
+});
+
+// Exporting as a loader from `preview.js` forces the promise to be resolved before EVERY story
+// See: https://storybook.js.org/docs/react/writing-stories/loaders
+export const loaders = isChromatic() && document.fonts ? [fontLoader] : [];
```

Comment:
```suggestion

// Use the document.fonts API to ensure fonts have fully loaded before rendering a story. Important for consistency when doing visual diffing in Chromatic.
// See: https://developer.mozilla.org/en-US/docs/Web/API/Document/fonts
const fontLoader = async () => ({
  fonts: await document.fonts.ready,
});

// Exporting as a loader from `preview.js` forces the promise to be resolved before EVERY story
// See: https://storybook.js.org/docs/react/writing-stories/loaders
export const loaders = document.fonts ? [fontLoader] : [];
```

Also force fonts to load on local dev, not just Chromatic.